### PR TITLE
Grouped related source code in sub-folders

### PR DESCRIPTION
**Pull request by duncanpMS**
No source code changes; no namespace changes. Existing tests pass.

Groups related source files into folders e.g. there is a folder that contains the platform-specific behaviour.
This makes it easier to see the decomposition of concepts within the project, which in turn makes it easier to understand and maintain.

### CODE REVIEW

**Comment by kpanwar** [comment]
public classes should be kept in the root namespace. As we move internal classes to folders, we should also update their namespaces.

File: src/ADAL.PCL/ADAL.PCL.csproj [hunk 1]
```diff
@@ -60,84 +60,85 @@
...
-    <Compile Include="TokenCacheKey.cs" />
-    <Compile Include="TokenCacheNotificationArgs.cs" />
-    <Compile Include="UserAssertion.cs" />
-    <Compile Include="UserCredential.cs" />
-    <Compile Include="UserIdentifier.cs" />
-    <Compile Include="UserInfo.cs" />
-    <Compile Include="UserRealmDiscoveryResponse.cs" />
-    <Compile Include="WsTrustRequest.cs" />
-    <Compile Include="WsTrustResponse.cs" />
+    <Compile Include="Caching\TokenCache.cs" />
```
Comment (kpanwar) [inline src/ADAL.PCL/ADAL.PCL.csproj]: consider renaming Caching to Cache.

File: src/ADAL.PCL/ADAL.PCL.csproj [hunk 2]
```diff
@@ -60,84 +60,85 @@
...
     <Compile Include="AuthenticationContext.cs" />
-    <Compile Include="AuthenticationParameters.cs" />
-    <Compile Include="AuthenticationResult.cs" />
-    <Compile Include="Authenticator.cs" />
-    <Compile Include="AuthenticatorTemplate.cs" />
+    <Compile Include="Data\AuthenticationParameters.cs" />
+    <Compile Include="Data\AuthenticationResult.cs" />
+    <Compile Include="Data\Authenticator.cs" />
+    <Compile Include="Data\AuthenticatorTemplate.cs" />
     <Compile Include="AuthenticatorTemplateList.cs" />
```
Comment (kpanwar) [inline src/ADAL.PCL/ADAL.PCL.csproj]: AuthenticatorTemplateList should also be in Authority folder.

File: src/ADAL.PCL/ADAL.PCL.csproj [hunk 3]
```diff
@@ -60,84 +60,85 @@
...
+    <Compile Include="Exceptions\AdalUserMismatchException.cs" />
     <Compile Include="AuthenticationContext.cs" />
-    <Compile Include="AuthenticationParameters.cs" />
-    <Compile Include="AuthenticationResult.cs" />
-    <Compile Include="Authenticator.cs" />
-    <Compile Include="AuthenticatorTemplate.cs" />
+    <Compile Include="Data\AuthenticationParameters.cs" />
+    <Compile Include="Data\AuthenticationResult.cs" />
+    <Compile Include="Data\Authenticator.cs" />
+    <Compile Include="Data\AuthenticatorTemplate.cs" />
```
Comment (kpanwar) [inline src/ADAL.PCL/ADAL.PCL.csproj]: consider renaming /Data to /Authority. Also, authenticantionparameters and authenticationresult do not belong to the /Data.

Comment (duncanpMS) [inline src/ADAL.PCL/ADAL.PCL.csproj]: Renamed

File: src/ADAL.PCL/ADAL.PCL.csproj (hunk 2 above)
Comment (duncanpMS) [inline src/ADAL.PCL/ADAL.PCL.csproj]: Done

File: src/ADAL.PCL/ADAL.PCL.csproj (hunk 1 above)
Comment (duncanpMS) [inline src/ADAL.PCL/ADAL.PCL.csproj]: Done

**Comment by duncanpMS** [comment]
I haven't changed any of the namespaces as just organising the code into related folders realises most of the value (and it avoid git getting confused by combining file moves and file edits). Changing the namespaces can be done separately if we think it's worth the effort.

**Pull request merged**